src/deadline/client/ui/dialogs/submit_job_to_deadline_dialog.py

```diff
@@ -400,6 +411,18 @@ def on_submit(self):
 
         # Submit the job
         try:
+            # Execute any PreSubmission function defined.
+            if self.on_pre_submit_callback:
+                self.on_pre_submit_callback(
+                    self,
+                    job_history_bundle_dir,
```

Comment:
This isn't initialized until a little bit further down, so it's currently undefined.

Reply:
I can move the pre_submit callback lower down so that the requirements are included and have it within the "show_host_requirements_tab" branch if that is recommended to be included.